### PR TITLE
Add Nancy to architect image

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/8901

Adds the `nancy` binary to the architect image to be used by `architect-orb`

### CODE REVIEW

**Comment by stone-z** [comment]
Testing on this `release-operator` PR https://github.com/giantswarm/release-operator/pull/189

File: Dockerfile [hunk 1]
```diff
@@ -37,7 +38,9 @@ RUN apk add --no-cache \
         curl -SL https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz | \
             tar -C /usr/bin --strip-components 1 -xvzf - linux-amd64/helm && \
         curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
-            sh -s -- -b $GOPATH/bin ${GOLANGCI_LINT_VERSION}
+            sh -s -- -b $GOPATH/bin ${GOLANGCI_LINT_VERSION} && \
+        curl -sSL -o /usr/bin/nancy https://github.com/sonatype-nexus-community/nancy/releases/download/${NANCY_VERSION}/nancy-linux.amd64-${NANCY_VERSION} && \
```
Comment (MarcelMue) [inline Dockerfile:42]: Can we somehow ensure that this binary doesn't move?

Not a big concern though.

Comment (tfussell) [reply Dockerfile:42]: It's not ideal, but it seems as safe as the other binaries above at least.

Comment (stone-z) [reply Dockerfile:42]: I'd certainly be open to a permalink or something, but I haven't seen one for the nancy releases

Comment (kopiczko) [reply Dockerfile:42]: Maybe checking the checksum would be a thing. If possible at all.

**Pull request merged**